File: client1/RunX/assets/scripts/app/PackageUtils.ts
/*
 * @Author: fly
 * @Date: 2021-03-16 11:13:51
 * @LastEditTime: 2021-03-16 13:43:01
 * @Description: 原生接口相关
 */

const PUClassName = "org.cocos2d.RunX.PackageUtils";

function isIos () {
    return cc.sys.os == cc.sys.OS_IOS;
}

function isAndroid () {
    return cc.sys.os == cc.sys.OS_ANDROID;
}

function isNative () {
    return cc.sys.isNative
}

class PackageUtils {
    isUseWifi (): boolean {
        if (isAndroid()) {
            //@ts-ignore
            return jsb.reflection.callStaticMethod(PUClassName, "IsUseWIFI", "()Z");
        } else if (isIos()) {
            // @ts-ignore
            return jsb.reflection.callStaticMethod("PackageUtils", "IsUseWIFI");
        } else {
            return true;
        }
    }

    getBatteryLevel (): number {
        if (isAndroid()) {
            //@ts-ignore
            return jsb.reflection.callStaticMethod(PUClassName, "GetBatteryLevel", "()D");
        } else if (isIos()) {
            // @ts-ignore
            return jsb.reflection.callStaticMethod("PackageUtils", "GetBatteryLevel");
        } else {
            return 1;
        }
    }

    getDeviceID (): string {
        if (isAndroid()) {
            //@ts-ignore
            return jsb.reflection.callStaticMethod(PUClassName, "GetDeviceID", "()Ljava/lang/String;");
        } else if (isIos()) {
            // @ts-ignore
            return jsb.reflection.callStaticMethod("PackageUtils", "GetDeviceID");
        } else {
            return "none";
        }
    }

    getPackageVersion (): string {
        if (isAndroid()) {
            //@ts-ignore
            return jsb.reflection.callStaticMethod(PUClassName, "GetPackageVersion", "()Ljava/lang/String;")
        } else if (isIos()) {
            // @ts-ignore
            return jsb.reflection.callStaticMethod("PackageUtils", "GetPackageVersion");
        } else {
            return "0.0.0"
        }
    }

    getPackageBuild (): string {
        if (isAndroid()) {
            //@ts-ignore
            return jsb.reflection.callStaticMethod(PUClassName, "GetPackageBuild", "()Ljava/lang/String;")
        } else if (isIos()) {
            // @ts-ignore
            return jsb.reflection.callStaticMethod("PackageUtils", "GetPackageBuild");
        } else {
            return "0"
        }
    }

    getIntentPath () {
        if (isAndroid()) {
            //@ts-ignore
            return jsb.reflection.callStaticMethod(PUClassName, "GetIntentPath", "()Ljava/lang/String;")
        } else if (isIos()) {
            // @ts-ignore
            return jsb.reflection.callStaticMethod("PackageUtils", "GetIntentPath")
        }
    }

    getIntentSearch () {
        if (isAndroid()) {
            //@ts-ignore
            return jsb.reflection.callStaticMethod(PUClassName, "GetIntentSearch", "()Ljava/lang/String;")
        } else if (isIos()) {
            // @ts-ignore
            return jsb.reflection.callStaticMethod("PackageUtils", "GetIntentSearch")
        }
    }

    getSdkInitStatus () {
        if (isAndroid()) {
            //@ts-ignore
            return jsb.reflection.callStaticMethod(PUClassName, "GetSdkInitStatus", "()I")
        } else if (isIos()) {
            // @ts-ignore
            return jsb.reflection.callStaticMethod("PackageUtils", "GetSdkInitStatus")
        }
    }

    getIsAntiAddiction () {
        if (isAndroid()) {
            //@ts-ignore
            return jsb.reflection.callStaticMethod(PUClassName, "GetIsAntiAddiction", "()V")
        } else if (isIos()) {
            // @ts-ignore
            return jsb.reflection.callStaticMethod("PackageUtils", "GetIsAntiAddiction")
        }
    }

    getAccountAge () {
        if (isAndroid()) {
            //@ts-ignore
            return jsb.reflection.callStaticMethod(PUClassName, "GetAccountAge", "()Ljava/lang/String;")
        } else if (isIos()) {
            // @ts-ignore
            return jsb.reflection.callStaticMethod("PackageUtils", "GetAccountAge")
        } else {
            // 未实名
            return "-1";
        }
    }

    changeAccount () {
        if (isAndroid()) {
            //@ts-ignore
            return jsb.reflection.callStaticMethod(PUClassName, "ChangeAccount", "()V")
        } else if (isIos()) {
            // @ts-ignore
            return jsb.reflection.callStaticMethod("PackageUtils", "ChangeAccount")
        }
    }

    login() {
        if (isAndroid()) {
            //@ts-ignore
            jsb.reflection.callStaticMethod(PUClassName, "Login", "()V")
        } else if (isIos()) {
            // @ts-ignore
            jsb.reflection.callStaticMethod("PackageUtils", "Login")
        }
    }

    logout () {
        if (isAndroid()) {
            //@ts-ignore
            jsb.reflection.callStaticMethod(PUClassName, "Logout", "()V")
        } else if (isIos()) {
            // @ts-ignore
            jsb.reflection.callStaticMethod("PackageUtils", "Logout")
        }
    }

    nativePay (order: string, productID: string, productName: string, price: number, 
        idealPrice: number, count: number, desc: string, orderTime: string, 
        accid: string, account: string, roleID: string, roleName: string, 
        level: string, serverID: string, serverName: string, ext: string, 
        session: string, sign: string, platOrderID: string, noticeUrl: string)
    {
        if (isAndroid()) {
            //@ts-ignore
            jsb.reflection.callStaticMethod(PUClassName, 
                "Pay", 
                "(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;I\
IILjava/lang/String;Ljava/lang/String;\
Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;\
Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;\
Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)V",
                order, productID, productName, price, 
                    idealPrice, count, desc, orderTime,
                    accid, account, roleID, roleName,
                    level, serverID, serverName, ext,
                    session, sign, platOrderID, noticeUrl)
        } else if (isIos()) {
            // @ts-ignore
            jsb.reflection.callStaticMethod("PackageUtils", "Pay")
        }
    }

    reportCreateRole (serverID: string, serverName: string, roleID: number, roleName: string, createTime: string) {
        if (isAndroid()) {
            //@ts-ignore
            jsb.reflection.callStaticMethod(PUClassName, 
                "CreateRole", 
                "(Ljava/lang/String;Ljava/lang/String;ILjava/lang/String;Ljava/lang/String;)V", 
                serverID, serverName, roleID, roleName, createTime)
        } else if (isIos()) {
            // @ts-ignore
            jsb.reflection.callStaticMethod("PackageUtils", "CreateRole:", roleID)
        }
    }

    reportLogin (serverID: string, serverName: string, roleID: number, roleName: string, level: number, createTime: string) {
        if (isAndroid()) {
            //@ts-ignore
            jsb.reflection.callStaticMethod(PUClassName, 
                "EnterGame", 
                "(Ljava/lang/String;Ljava/lang/String;ILjava/lang/String;ILjava/lang/String;)V",
                serverID, serverName, roleID, roleName, level, createTime)
        } else if (isIos()) {
            // @ts-ignore
            jsb.reflection.callStaticMethod("PackageUtils", "EnterGame:", roleID)
        }
    }

    reportLevelUp (serverID: string, serverName: string, roleID: number, roleName: string, level: number, createTime: string) {
        if (isAndroid()) {
            //@ts-ignore
            jsb.reflection.callStaticMethod(PUClassName, 
                "UserLevelUp", 
                "(Ljava/lang/String;Ljava/lang/String;ILjava/lang/String;ILjava/lang/String;)V",
                serverID, serverName, roleID, roleName, level, createTime)
        } else if (isIos()) {
            // @ts-ignore
            jsb.reflection.callStaticMethod("PackageUtils", "EnterGame:", roleID)
        }
    }
}

let packageUtils = new PackageUtils()
export {
    packageUtils as default,
    isIos,
    isAndroid,
    isNative
}

File: client1/RunX/build-templates/jsb-default/frameworks/runtime-src/proj.android-studio/app/src/org/cocos2d/RunX/PackageUtils.java
package org.cocos2d.RunX;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import org.cocos2dx.javascript.AppActivity;
import org.json.JSONArray;
import org.json.JSONObject;

import com.talkingsdk.models.PayData;
import com.talkingsdk.models.PlayerData;
import com.zqgame.RunX.BuildConfig;
import com.talkingsdk.MainApplication;

import java.util.HashMap;
import java.util.Map;

@SuppressWarnings("unused")
public class PackageUtils {

    static boolean IsUseWIFI () {
        ConnectivityManager cm = (ConnectivityManager) AppActivity.getContext().getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo info = cm.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
        return info.isConnected();
    }

    static double GetBatteryLevel () {
        return 0;
    }

    static String GetDeviceID () {
        return "none";
    }

    static String GetPackageVersion () {
        return BuildConfig.VERSION_NAME;
    }

    static String GetPackageBuild () {
        return Integer.toString(BuildConfig.VERSION_CODE);
    }

    static String GetIntentPath () {
        return AppActivity.intentPath;
    }

    static String GetIntentSearch () {
        return AppActivity.intentSearch;
    }

    static int GetSdkInitStatus () {
        return AppActivity.sdkStatus;
    }

    static String GetAccountAge () {
        return AppActivity.age;
    }

    static void Login () {
        MainApplication.getInstance().login();
    }

    static void LoginWX4YYB () {
        MainApplication.getInstance().login(35);
    }

    static void LoginQQ4YYB () {
        MainApplication.getInstance().login(27);
    }

    static void Logout () {
        MainApplication.getInstance().logout();
    }

    static void GetIsAntiAddiction () {
        MainApplication.getInstance().getIsAntiAddiction();
    }

    static void ChangeAccount () {
        MainApplication.getInstance().changeAccount();
    }

    static void CreateRole (String serverID, String serverName, int roleID, String roleName, String createTime) {
        PlayerData _playerData = new PlayerData();

        _playerData.setServerNo(serverID);
        _playerData.setServerName(serverName);
        _playerData.setRoleId(roleID);
        _playerData.setRoleName(roleName);
        _playerData.setLevel(1);

        Map<String,String> exs = new HashMap<String,String>();
        exs.put("roleCTime",createTime);// 单位：秒 即10位数
        _playerData.setEx(exs);

        MainApplication.getInstance().createRole(_playerData);
    }

    static void EnterGame (String serverID, String serverName, int roleID, String roleName, int level, String createTime) {
        PlayerData _playerData = new PlayerData();

        _playerData.setServerNo(serverID);
        _playerData.setServerName(serverName);
        _playerData.setRoleId(roleID);
        _playerData.setRoleName(roleName);
        _playerData.setLevel(level);

        Map<String,String> exs = new HashMap<String,String>();
        exs.put("roleCTime",createTime);// 单位：秒 即10位数
        _playerData.setEx(exs);

        MainApplication.getInstance().enterGame(_playerData);
    }

    static void UserLevelUp (String serverID, String serverName, int roleID, String roleName, int level, String createTime) {
        PlayerData _playerData = new PlayerData();

        _playerData.setServerNo(serverID);
        _playerData.setServerName(serverName);
        _playerData.setRoleId(roleID);
        _playerData.setRoleName(roleName);
        _playerData.setLevel(level);

        Map<String,String> exs = new HashMap<String,String>();
        exs.put("roleCTime",createTime);// 单位：秒 即10位数
        _playerData.setEx(exs);

        MainApplication.getInstance().userUpLevel(_playerData);
    }

    static void Pay (
            String order,
            String productID,
            String productName,
            int price,
            int idealPrice,
            int count,
            String desc,
            String orderTime,
            String accid,
            String account,
            String roleID,
            String roleName,
            String level,
            String serverID,
            String serverName,
            String ext,
            String session,
            String sign,
            String platOrderID,
            String noticeUrl
    ) {
        PayData payData = new PayData();

        payData.setSubmitTime(orderTime);
        payData.setMyOrderId(order);
        payData.setDescription(desc);
        payData.setProductCount(count);
        payData.setProductId(productID);
        payData.setProductIdealPrice(price);
        payData.setProductRealPrice(price);
        payData.setProductName(productName);
        
        Map<String,String> exs = new HashMap<String,String>();

        exs.put("UserLevel", level);
        exs.put("UserRoleName", roleName);
        exs.put("UserRoleId", roleID);
        exs.put("UserServerName", serverName);
        exs.put("UserServerId", serverID);
        exs.put("GameMoneyAmount", String.valueOf(count));
        exs.put("GameMoneyName", productName);
        exs.put("UserBalance", "0");
        exs.put("UserGamerVip", "0");
        exs.put("UserPartyName", "无");

        exs.put("UserId", accid);
        exs.put("LoginAccount", account);
        exs.put("LoginDataEx", ext);
        exs.put("LoginSession", session);
        exs.put("AccessKey", sign);
        exs.put("OutOrderID", platOrderID);
        exs.put("NoticeUrl", noticeUrl);

        payData.setEx(exs);
        MainApplication.getInstance().pay(payData);
    }
}
